packages/devtools_app/test/support/mocks.dart

```diff
@@ -49,11 +50,11 @@ class FakeServiceManager extends Fake implements ServiceConnectionManager {
       FakeServiceExtensionManager();
 
   @override
-  StreamSubscription<bool> hasRegisteredService(
+  ValueListenable<bool> getRegisteredServiceListenable(
     String name,
-    void onData(bool value),
   ) {
-    return Stream.value(false).listen(onData);
+    final ValueListenable listenable = ImmediateValueNotifier(false);
```

Comment:
[nit] Why do we need this type annotation?

How about


`return ImmediateValueNotifier(false);`?

Reply:
I want to return a ValueListenable so that listeners can't change the value of the ValueNotifier - only read it.

Reply:
That's controlled by the return type on the method signature, not by a local variable type within the function.